fairseq/criterions/wav2vec_criterion.py

```diff
@@ -90,6 +90,7 @@ def forward(self, model, sample, reduce=True, log_pred=False):
             "ntokens": sample_size,
             "nsentences": sample["id"].numel(),
             "sample_size": sample_size,
+            **model.get_alignment_metrics(**sample["net_input"], alignments=sample["alignments"])
```

Comment:
If we do it like that, we will slow down training by ca 1.5x, because the metrics redo work (recompute features etc.)

Better would be to:
- add this computation to model's forward and stash it in `net_output`
- - fetch it here from `net_output`

In that way we are not redoing computations

Reply:
That's on my TODO list.

We could improve speed further by re-using quantization computed during the `forward` method - however, we only quantize the unmasked features. We would not compute metrics on masked indices by doing it this way.

A decision between those two is to be made.